templates/components/form/fields_macros.html.twig

```diff
@@ -122,13 +122,17 @@
 
 {% macro sliderField(name, value, label = '', options = {}) %}
    {% if options.fields_template.isMandatoryField(name) %}
-      {% set options = {'required': true}|merge(options) %}
+      {% set options = {
+         'no_value': 0,
+         'yes_value': 1,
+         'required': true
+      }|merge(options) %}
    {% endif %}
 
    {% set field %}
       <label class="form-check form-switch pt-2">
-         <input type="hidden"   name="{{ name }}" value="0" />
-         <input type="checkbox" name="{{ name }}" value="1" class="form-check-input" id="%id%"
+         <input type="hidden"   name="{{ name }}" value="{{ options.no_value }}" />
+         <input type="checkbox" name="{{ name }}" value="{{ options.yes_value }}" class="form-check-input" id="%id%"
                 {{ value == 1 ? 'checked' : '' }}
```

Comment:
I wonder if it would not be more logical to compage agains the yes_value
```suggestion
                {{ value == options.yes_value ? 'checked' : '' }}
```

Reply:
I think it makes sense as-is. The `value` should be 0 or 1 to indicate the state of the input while `no_value` and `yes_value` indicate the actual value depending on the state of the input.

The way it seems to work now is the no/default value of this control is 0 for the KB ID while the yes value is the actual KB ID. If the toggle is off, a 0 is submitted for the ID. If the toggle is on, the actual ID is submitted. I don't know why it was done that way instead of a regular toggle/checkbox with a separate flag, but it was like that before 10.0.

Reply:
This fix create other error SQL:

[2022-09-12 16:37:01] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi/src/DBmysql.php line 370
  *** MySQL query error:
  SQL: INSERT INTO `glpi_itilfollowups` (`itemtype`, `items_id`, `content`, `requesttypes_id`, `is_private`, `users_id`, `timeline_position`, `date`, `date_creation`, `date_mod`) VALUES ('Ticket', '9218', 'comment;', '0', '', '1773', '1', '2022-09-12 16:37:01', '2022-09-12 16:37:01', '2022-09-12 16:37:01')
  Error: Incorrect integer value: '' for column `glpi`.`glpi_itilfollowups`.`is_private` at row 1
  Backtrace :
  src/DBmysql.php:1312                               
  src/CommonDBTM.php:705                             DBmysql->insert()
  src/CommonDBTM.php:1304                            CommonDBTM->addToDB()
  front/itilfollowup.form.php:55                     CommonDBTM->add()

Reply:
Did you make the changes to the default options a few lines above this?

Reply:
I changes all lines about this fix https://github.com/glpi-project/glpi/pull/12624/commits/28baebe4c7809bb8fef023d9ea6bde62858cf7d2

And Suggested change from @cedric-anne 

            -    {{ value == 1 ? 'checked' : '' }}
            +    {{ value == options.yes_value ? 'checked' : '' }}

Reply:
The new commit should fix that issue.
f03949bec1b9083ac82ee855226866c41be8de29